device/core/src/client.ts

```diff
@@ -469,6 +552,35 @@ export class Client extends EventEmitter {
     }
   }
 
+  private _enableMethods(callback: (err?: Error) => void): void {
+    if (!this._methodsEnabled) {
+      const retryOp = new RetryOperation(this._retryPolicy, this.maxOperationTimeout);
+      retryOp.retry((opCallback) => {
+      this._transport.enableMethods(opCallback);
+      }, (err) => {
+        if (!err) {
+          this._methodsEnabled = true;
+        }
+        callback(err);
+      });
+    } else {
+      callback();
+    }
+  }
+
+  // private _disableMethods(callback: (err?: Error) => void): void {
```

Comment:
>// [](start = 2, length = 2)

?

Reply:
added a clarifying comment in iteration 3

---
In reply to: [146323974](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r146323974) [](ancestors = 146323974)